### PR TITLE
.Net: [dev] Improve KernelSyntaxExamples ability to filter examples.

### PR DESCRIPTION
**Pull request by lemillermicrosoft**
### Motivation and Context

Make it easier to run individual examples in the KernelSyntaxExamples project.

### Description
This commit combines the updates to launch.json, tasks.json, and
Program.cs to allow passing arguments to the main method and running
examples based on the provided filter. The list of examples in
Program.cs has also been updated to include new examples and ensure
proper execution. These changes make it easier to run specific examples
without having to run all of them and improve the organization and
readability of the code.

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

**Review by hario90 — APPROVED** [review]

File: dotnet/samples/KernelSyntaxExamples/Program.cs [hunk 1]
```diff
@@ -19,56 +20,85 @@ public static async Task Main()
...
+                    Console.WriteLine($"Running {example.Key}...");
+                    await example.Value(cancellationToken).SafeWaitAsync(cancellationToken);
+                }
+                catch (ConfigurationNotFoundException ex)
+                {
+                    Console.WriteLine($"{ex.Message}. Skipping example {example.Key}.");
+                }
+            }
+        }
     }
```
Comment (RogerBarreto) [inline dotnet/samples/KernelSyntaxExamples/Program.cs:102]: Using reflection you can massively simplify this implementation.

```suggestion
    private static async Task RunExamplesAsync(string? filteredExample, CancellationToken cancellationToken)
{
    var examplesToRun = (Assembly.GetExecutingAssembly().GetTypes())
        .Where(type => type.Name.StartsWith("Example", StringComparison.OrdinalIgnoreCase))
        .Select(type => type.Name).ToList();

      if (!string.IsNullOrWhiteSpace(filteredExample))
      {
          if (!examplesToRun.Contains(filteredExample))
          {
              Console.WriteLine($"Example \"{filteredExample}\" not found.");
              Console.WriteLine($"\nUse one of the available examples below:\n\n{string.Join('\n', examplesToRun)}");
              return;
          }
  
          // Clean up other examples
          examplesToRun.Clear();
  
          // Add selected one
          examplesToRun.Add(filteredExample);
      }
  
      foreach (var example in examplesToRun)
      {
          await RunSafeExampleAsync(example, cancellationToken);
      }
  }
  
  private static async Task RunSafeExampleAsync(string exampleClass, CancellationToken cancellationToken)
{
    var invokeExample = () =>
    {
        var method = Assembly.GetExecutingAssembly().GetType(exampleClass)?.GetMethod("RunAsync")
                     ?? throw new ArgumentOutOfRangeException(nameof(exampleClass), $"Example {exampleClass} not found");

        method.Invoke(null, null);
        return Task.CompletedTask;
    };

    try
    {
        Console.WriteLine($"Running {exampleClass}...");
        await SafeWaitAsync(invokeExample(), cancellationToken);
    }
    catch (ConfigurationNotFoundException ex)
    {
        Console.WriteLine($"{ex.Message}. Skipping example {exampleClass}.");
    }
}
```

Comment (lemillermicrosoft) [reply dotnet/samples/KernelSyntaxExamples/Program.cs:102]:
```
string? filter = args.Length > 0 ? args[0] : null;
```
Option 1: Change the `null` here to your default filter.

Option 2: Set Command-Line Arguments:

Right-click on your console application project in the Solution Explorer.
Choose "Properties" from the context menu.
In the project properties window, navigate to the "Debug" tab on the left.
Supply Command-Line Arguments:

In the "Command line arguments" field, enter the command-line arguments that your console application expects. Separate multiple arguments with spaces.

Comment (hario90) [reply dotnet/samples/KernelSyntaxExamples/Program.cs:102]: Can you add these instructions to the README for this project?

**Review by RogerBarreto — APPROVED** [review]
Great improvement. LGTM

**Pull request merged**